### PR TITLE
Add on-device runners

### PR DESCRIPTION
**Pull request by jchalatu**
Adding on-device runners to nengo-edge for Coral, Nordic and Discovery boards. Introduces new functionality for loading `parameters.json` artifact that accompanies exported models from NengoEdge UI. Also removed ability for users to disable feature extraction when interfacing with a runner. 

Two example notebooks are added to provide users with a template on how to interact with the new runners for their relevant device(s).

### CODE REVIEW

File: nengo_edge/tests/test_micro_runner.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        n_unroll=1,
+        monkeypatch=monkeypatch,
+    )
+
+    inputs = rng.uniform(-0.5, 0.5, size=(batch_size, 10))
+
+    # check sample rate too low
+    tflmRunner.preprocessing["sample_rate"] = 10
+    with tflmRunner as runner:
+        with pytest.raises(RuntimeError):
```
Comment (drasmuss) [inline nengo_edge/tests/test_micro_runner.py:211]: Should have a `match=...` parameter here.

File: nengo_edge/tests/test_micro_runner.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            rxd_data = np.ones((n_unroll * (size // n_unroll)), dtype="float32")
+        else:
+            rxd_data = np.ones((size), dtype="float32")
+
+        return rxd_data
+
+    monkeypatch.setattr(MicroRunner, "__enter__", mock_enter)
+    monkeypatch.setattr(MicroRunner, "__exit__", mock_exit)
+    monkeypatch.setattr(MicroRunner, "send_data_burst", mock_send_data_burst)
+    monkeypatch.setattr(MicroRunner, "receive_data_burst", mock_receive_data_burst)
```
[the remark this answers is not in the document]
Comment (drasmuss) [reply nengo_edge/tests/test_micro_runner.py:90]: After reworking MicroRunner to use Disco/NordicRunner, and no longer mocking enter/exit, it made more sense to just leave it this way.

File: nengo_edge/network_runner.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        self.prepared = False
+
+    def send_inputs(self, inputs: np.ndarray) -> None:
+        """Saves inputs to file and sends to device."""
+
+        # save inputs to file
+        with zipfile.ZipFile(self.directory / "inputs.zip", "w") as f:
+            np_bytes = io.BytesIO()
+            np.save(np_bytes, inputs, allow_pickle=False)
+            f.writestr("inputs", np_bytes.getvalue())
```
[the remark this answers is not in the document]
Comment (drasmuss) [reply nengo_edge/network_runner.py]: I changed it to npz and it seems to work so 🤷

**Review by jchalatu — COMMENTED** [review]
Fixups LGTM. Will takeover making the final changes from last commit.

**Review by drasmuss — APPROVED** [review]

**Pull request merged**